bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/graphics/ImageLoader.java

```diff
@@ -291,18 +344,43 @@ boolean isInterlacedPNG(byte [] imageAsByteArray) {
  */
 public ImageData[] load(String filename) {
 	if (filename == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
-	InputStream stream = null;
-	try {
-		stream = new FileInputStream(filename);
-		return load(stream);
```

Comment:
Please submit the simplification of the existing `ImageData[] load(String filename)` method (i.e. using try-with-resources in Linux and Mac, Windows seems already cleaned-up) as a separate PR.
This way we can reduce the number of not directly related changes in this PR.